jobs/kibana/templates/config/kibana.conf.erb

```diff
@@ -5,35 +5,56 @@ port: <%= p('kibana.port') %>
 host: "0.0.0.0"
 
 # The Elasticsearch instance to use for all your queries.
-elasticsearch: "<%= p('kibana.elasticsearch') %>"
+elasticsearch_url: "http://<%= p('kibana.api_host') %>"
+
+# preserve_elasticsearch_host true will send the hostname specified in `elasticsearch`. If you set it to false,
+# then the host you use to connect to *this* Kibana instance will be sent.
+elasticsearch_preserve_host: true
 
 # Kibana uses an index in Elasticsearch to store saved searches, visualizations
 # and dashboards. It will create a new index if it doesn't already exist.
 kibana_index: "<%= p('kibana.kibana_index') %>"
 
+# If your Elasticsearch is protected with basic auth, this is the user credentials
+# used by the Kibana server to perform maintence on the kibana_index at statup. Your Kibana
+# users will still need to authenticate with Elasticsearch (which is proxied thorugh
+# the Kibana server)
+# kibana_elasticsearch_username: user
+# kibana_elasticsearch_password: pass
+
 # The default application to load.
 default_app_id: "<%= p('kibana.default_app_id') %>"
 
-# Time in seconds to wait for responses from the back end or elasticsearch.
-# Note this should always be higher than "shard_timeout".
+# Time in milliseconds to wait for responses from the back end or elasticsearch.
 # This must be > 0
-request_timeout: 60
+request_timeout: 300000
```

Comment:
These shard and request timeouts are heavily data and cluster dependent. I think they should be configurable (and defaulted to kibana's built-in defaults, if they're not already).